src/SimpleDb.php

```diff
@@ -9,28 +9,30 @@
  */
 class SimpleDb
 {
-    /** @var AdapterInterface */
+    /** @var array */
     private $adapter;
-
-    /**
-     * @param AdapterInterface $adapter
-     */
     public function __construct(AdapterInterface $adapter)
     {
         $this->adapter = $adapter;
     }
 
-    /**
-     * @return array
-     */
-    public function findAll()
+    public function findAll(array $criterias = array())
     {
-        $data = $this->adapter->read();
+        $filteredData = array();
+        $rawData =  $this->adapter->read();
+
+        if (!$criterias) {
+            return $rawData;
+        }
 
-        if (isset($data['data'])) {
-            return $data['data'];
+        foreach ($rawData as $record) {
+            foreach ($criterias as $field  => $value) {
+                if ($record[$field] == $value) {
```

Comment:
This will cause duplication if more than one criteria is sent. Why not to create a test that will expose the bug and then fix it?